lib/rubocop/cop/rails/enum_hash.rb

```diff
@@ -30,6 +30,35 @@ def on_send(node)
             add_offense(node, message: format(MSG, enum: name))
           end
         end
+
+        def autocorrect(node)
+          enum_values = node.children[2].children.first.children[1]
+          to_replace = enum_values.loc.expression
+          values_hash = "{ #{converted_values(enum_values)} }"
+
+          ->(corrector) { corrector.replace(to_replace, values_hash) }
+        end
+
+        private
+
+        def converted_values(enum_values)
+          enum_values.children.each_with_index.map do |child, index|
+            hash_entry_as_string(child, index)
+          end.join(', ')
+        end
+
+        def hash_entry_as_string(child, index)
+          value = child.children.first
+          case value
+          when String
+            "'#{value}' => #{index}"
```

Comment:
Use `String#dup` instead of `"'#{value}'"`. 
https://docs.ruby-lang.org/en/2.3.0/String.html#method-i-dump

`eval(any_string.dump)` always eauqlas `any_string`.

Reply:
I did it this way to prevent escaped keys, for example: 
`got: "enum status: {:old=>0, :\"very active\"=>1, \"is archived\"=>2, 42=>3}\n"`

if that's not a problem would it be fine to just build a real hash and stringify it? It'd solve the problem being discussed here https://github.com/rubocop-hq/rubocop-rails/pull/95#discussion_r307575556